src/pygit2.c

```diff
@@ -183,13 +183,34 @@ init_file_backend(PyObject *self, PyObject *args)
 }
 
 
+PyDoc_STRVAR(reference_is_valid_name__doc__,
+    "reference_is_valid_name(refname) -> bool\n"
+    "\n"
+    "Check if the passed string is a valid reference name.");
+PyObject *
+reference_is_valid_name(PyObject *self, PyObject *args)
+{
+    char* refname;
+    int result;
+
+    if (!PyArg_ParseTuple(args, "es", "utf-8", &refname)) {
```

Comment:
See [Python's documentation](https://docs.python.org/2/c-api/arg.html) for `"es"`. It allows you to pass in a Unicode string and encodes it as UTF-8.

This is not necessary on Python 3 because `"s"` converts to UTF-8. But in Python 2, it converts to the "default encoding". That makes the test fail like this:

```
>       assert reference_is_valid_name('refs/heads/ünicöde')
E       UnicodeEncodeError: 'ascii' codec can't encode character u'\xfc' in position 11: ordinal not in range(128)
```

Reply:
you may use `METH_O` instead of `METH_VARARGS` and then:

    refname = py_str_to_c_str(py_refname, NULL);

that will use utf-8 for Python 2 as well.